crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -77,13 +78,24 @@ impl Actor for C8yMapperActor {
     }
 
     async fn run(mut self) -> Result<(), RuntimeError> {
+        let log_if_bridge_not_detected = tokio::spawn(async {
+            tokio::time::sleep(Duration::from_secs(10)).await;
+            error!(
+                "No valid health status message has been detected for Cumulocity bridge for 10s, mapper has not yet started"
+            );
+        });
+
         // Wait till the c8y bridge is established
+        info!("Waiting for Cumulocity bridge to be established before mapper starts");
         while let Some(message) = self.bridge_status_messages.recv().await {
             if is_c8y_bridge_established(&message, &self.c8y_bridge_service_name) {
                 break;
             }
         }
 
+        log_if_bridge_not_detected.abort();
+        info!("Cumulocity bridge is established, mapper is starting");
```

Comment:
Having a background error printing task, that is aborted on the happy path, is more than unusual!

But that's not my main point.
I wonder, why introducing a timeout that was not present before?

Actually:
- with the mosquitto bridge, the c8y mapper *must* wait for the bridge to be established (up or down it doesn't matter) otherwise all the init messages sent on the `c8y/#` topics will be lost (as there is no persisted subscriber).
- with the built-in bridge, the c8y mapper *can* directly send the init messages, since there is one subscriber to the `c8y/#` topics (the mapper itself).

To make things simple, I would only take care of the case where awaiting the bridge is required.

Reply:
The timeout doesn't change the behaviour it exists purely so the mapper can log that it's totally not functional because the bridge doesn't exist (and I couldn't think of a better way to notify than just "if nothing has happened after a certain amount of time, something is broken"). I agree that this isn't necessary though, and I should just skip it if `c8y.bridge.built_in` is `true`.

Reply:
> I didn't consider it too deeply as I was mostly concerned with "this bit of code that's critically breaking the mapper didn't tell me it existed" and that taking a while to debug. 

That's true that some logging is required as the mapper is doing nothing this the bride is connected.

>  I should just skip it if `c8y.bridge.built_in` is `true`.

Yes, if this is actually simpler.

Reply:
I just saw https://github.com/thin-edge/thin-edge.io/pull/2810/commits/4ef0fcd5aa235612f682fb725cd8c031073ba4a6 and I'm okay with it.